chassis.go

```diff
@@ -32,66 +32,71 @@ type Chassis struct {
 	VtepLogicalSwitches []string
 }
 
-func (odbi *ovndb) chassisAddImp(name string, hostname string, etype string, ip string) (*OvnCommand, error) {
-	// / Prepare for encap record
-	enCapUUID, err := newRowUUID()
-	if err != nil {
-		return nil, err
-	}
-	row := make(OVNRow)
-
-	if len(name) > 0 {
-		row["chassis_name"] = name
-	}
-
+func (odbi *ovndb) chassisAddImp(name string, hostname string, etype []string, ip string) (*OvnCommand, error) {
+	//Prepare for encap record
+	var encap_ids [][]string
+	var namedUUID = "named-uuid"
+	var operations []libovsdb.Operation
 	if len(etype) > 0 {
-		row["type"] = etype
-	}
+		for _, et := range etype {
+			enCapUUID, err := newRowUUID()
+			if err != nil {
+				return nil, err
+			}
+			row := make(OVNRow)
 
-	if len(ip) > 0 {
-		row["ip"] = ip
-	}
-	if uuid := odbi.getRowUUID(tableEncap, row); len(uuid) > 0 {
-		return nil, ErrorExist
+			if len(name) > 0 {
+				row["chassis_name"] = name
+			}
+			var encap_id []string
+			encap_id = append(encap_id, namedUUID)
+			encap_id = append(encap_id, enCapUUID)
+			encap_ids = append(encap_ids, encap_id)
+			if len(ip) > 0 {
+				row["ip"] = ip
+			}
+			row["type"] = et
+			if uuid := odbi.getRowUUID(tableEncap, row); len(uuid) > 0 {
+				return nil, ErrorExist
+			}
+			insertEncapOp := libovsdb.Operation{
+				Op:       opInsert,
+				Table:    tableEncap,
+				Row:      row,
+				UUIDName: enCapUUID,
+			}
+			// check if chassis already exists with that encap
+			if uuid := odbi.getRowUUID(tableChassis, row); len(uuid) > 0 {
```

Comment:
How could a row of Encap exists in Chassis table? This whole "if" should be removed.

Reply:
My bad as this is mistake while refactor as this condition will never be true, UT didn't catch it. I will move this  out out encap loop and basically its checking if raw with uuid already exists in chassis table before creating a new one

Reply:
You generated a new UUID, there is no point to check if a new UUID existed. UUID is supposed to be universally unique.